picobot/video_editor.py

```diff
@@ -0,0 +1,46 @@
+from pathlib import Path
+import ffmpeg
+
+
+def sticker_from_video(mp4_path: Path):
+    '''
+    Converts the given video to a proper format that can be uploaded as a sticker.
+    Telegram accepts WEBM VP9 with a maximum size of 512x512 pixels with maximum 3 seconds duration.
+    '''
+    probe = ffmpeg.probe(mp4_path)
+    video_stream = next(
+        (stream for stream in probe['streams'] if stream['codec_type'] == 'video'), None
+    )
+    width = int(video_stream['width'])
+    height = int(video_stream['height'])
+    duration = float(probe['format']['duration'])
+
+    check_duration(duration)
+
+    (width, height) = estimate_video_sticker_size(width, height)
+    vid_output_path = mp4_path.with_suffix('.webm')
+    (
+        ffmpeg.input(mp4_path)
+        .filter('scale', width, height)
+        .output(vid_output_path.as_posix(), format='webm', vcodec='libvpx-vp9')
+        .run()
```

Comment:
[2] Trying to `/addsticker` a video a second time as the same user makes `ffpmeg` ask to overwrite the file:

```console
File '/home/jptiz/dev/car/pico-bot/picobot/images/vid<my user ID>.webm' already exists. Overwrite? [y/N] y
```

Maybe one of these may work, according to [ffmpeg-python docs](https://kkroening.github.io/ffmpeg-python/#ffmpeg.overwrite_output):

1. Adding `.overwrite_output()` to ffmpeg builder:
```suggestion
        .output(vid_output_path.as_posix(), format='webm', vcodec='libvpx-vp9')
        .overwrite_output()
        .run()
```


2. ***Or*** adding `overwrite_output` kwarg to `.run`:
```suggestion
        .output(vid_output_path.as_posix(), format='webm', vcodec='libvpx-vp9')
        .run(overwrite_output=True)
```